cmd/operator/main.go

```diff
@@ -54,19 +55,46 @@ func init() {
 
 type devicePluginControllerAndWebhook map[string](func(ctrl.Manager, string, bool) error)
 
+type flagList []string
+
+func (fl *flagList) String() string {
+	return strings.Join(*fl, ", ")
+}
+
+func (fl *flagList) Set(value string) error {
+	*fl = append(*fl, value)
+	return nil
+}
+
+func enabled(arr []string, val string) bool {
+	for _, s := range arr {
+		if s == val {
+			return true
+		}
+	}
+	return false
+}
+
 func main() {
+	var devices flagList
 	var metricsAddr string
 	var devicePluginNamespace string
 	var enableLeaderElection bool
+	var pm *patcher.PatcherManager
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&devicePluginNamespace, "deviceplugin-namespace", metav1.NamespaceSystem, "The namespace where deviceplugin daemonsets are created")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.Var(&devices, "devices", "Device(s) to set up.")
 	flag.Parse()
 
 	ctrl.SetLogger(klogr.New())
 
+	if len(devices) == 0 {
+		devices = []string{"dsa", "fpga", "gpu", "qat", "sgx"}
+	}
```

Comment:
this would not be needed as well if we use a map as we can initialise it with all supported plugins:
```
var devices = flagMap{"gpu": false, "fpga": false, "sgx": false, "dsa": false, "vpu": false}
```

Reply:
BTW, this approach would help to invalidate incorrect user input, e.g. if user uses "--device unsupported" we can throw an error.